python/graphstorm/model/utils.py

```diff
@@ -419,6 +494,48 @@ def save_embeddings(model_path, embeddings, local_rank, world_size,
         with open(os.path.join(model_path, "emb_info.json"), 'w', encoding='utf-8') as f:
             f.write(json.dumps(emb_info))
 
+def save_embeddings(model_path, embeddings, local_rank, world_size,
+    device=th.device('cpu'), node_id_mapping_file=None, 
+    save_embed_format="pytorch"):
+    """ Save embeddings.
+
+        Parameters
+        ----------
+        model_path : str
+            The path of the folder where the model is saved.
+        embeddings : DistTensor
+            Embeddings to save
+        local_rank : int
+            Local rank
+        world_size : int
+            World size in a distributed env.
+        device: torch device
+            Device used for all_to_allv data exchange. For gloo backend
+            we store data in CPU, For nccl backend, we need to store
+            data in GPU.
+        node_id_mapping_file : str
+            Path to the file storing node id mapping generated by the
+            graph partition algorithm.
+        save_embed_format : str
+            The format of saved embeddings. 
+            Currently support ["pytorch", "hdf5"].
+    """
+    os.makedirs(model_path, exist_ok=True)
+    if save_embed_format == "pytorch":
+        logging.info(f"Writing GNN embeddings to "\
+            f"{model_path} in pytorch format.")
+        save_pytorch_embeddings(model_path, embeddings, local_rank, world_size,
+            device, node_id_mapping_file)
+    elif save_embed_format == "hdf5":
+        mapped_embeds = remap_embeddings(embeddings, local_rank, world_size,
```

Comment:
It is better to write a save_hdf5_embeddings function and wrap L530 to L 535 in it.

Reply:
updated.